Themis/ViewModels/Assessment/AssessmentViewModel.swift

```diff
@@ -1,72 +1,20 @@
 import Foundation
 import SwiftUI
+import Combine
 
 class AssessmentViewModel: ObservableObject {
-    @Published var fileTree: [Node] = []
-    @Published var openFiles: [Node] = []
-    @Published var selectedFile: Node?
-    @Published var editorFontSize = CGFloat(14) // Default font size
-
-    init(files: [String: FileType]) {
-        let node = ArtemisAPI.initFileTreeStructure(files: files)
-        self.fileTree = node.children ?? []
-    }
-
-    func openFile(file: Node) {
-        if !openFiles.contains(where: { $0.path == file.path }) {
-            openFiles.append(file)
-            file.fetchCode()
+    @Published var submission: SubmissionForAssessment?
+    @Published var foundSubmission: Bool = false
```

Comment:
This could be a computed property. According to https://stackoverflow.com/a/62995401/4306257, it will also automatically update.
```suggestion
    val foundSubmission: Bool { submission != nil }
```

Reply:
Yes, definitely better

Reply:
nvm doesn't work, we obviously need the property wrapper in the SubmissionListView

Reply:
Oh, I see. But this also means that `foundSubmission` could be `false` even if the submission is not `nil` (because the navigation destination is closed again), which is really weird. Maybe change the variable name to something like `showSubmission`?

Reply:
I mean the foundSubmission is only to notify that the view should navigate. I can change the name to make it more obvious, yes